### DISCUSSION TITLE
Matching patterns

### DISCUSSION THREAD

**Discussion by cmutel** [cauldron/bw_matchbox#71] [General]
Match patterns string [are stored here](https://github.com/cauldron/bw_matchbox/blob/037e046e40924f6aaea312071fa2512044e12fd3/bw_matchbox/utils.py#L61) - this is also where to add new ones. Don't forget to [also add an abbreviation](https://github.com/cauldron/bw_matchbox/blob/037e046e40924f6aaea312071fa2512044e12fd3/bw_matchbox/webapp.py#L370)!

# 1. No direct match available

A new set of data was added with doesn't exist in the target database, or a different modelling approach was taken which doesn't match with the target.

Example: https://dare.cauldron.ch/match/26719

Action: *No match needed* button. We don't try to match incompatible datasets. Match type is created automatically.

# 2. Direct match with near-identical data

The same dataset is present in both databases, even if it's name might be slightly different.

Example: https://dare.cauldron.ch/process/27102

Action: *1-to-1 Proxy* button. We indicate that these datasets are directly compatible. Match type is created automatically.

# 3. Match with updated data

These datasets had the same origin, but the target process has been updated with new numbers or the addition of more inputs. However, they still are a direct replacement for each other.

Example: https://dare.cauldron.ch/process/29405

Action: *Collapse inputs* arrow, then *Create proxy* button. Match type "Match with updated data" should be selected.

# 4. Match with updated data and adding source transport

These datasets had the same origin, but the target process was remodelled and transport shifted to another (usually market) process. To best match the intent of the original dataset, its transport inputs are copied into the new proxy dataset.

Example: https://dare.cauldron.ch/compare/32751/9705

Actions: *Collapse inputs* arrow, then use "Shift row" arrows to copy over transport inputs. *Create proxy* button, and set match type to "Match with updated data and adding source transport".

# 5. Match with market and replace or remove transport

Sometimes the best match for a UVEK process is an ecoinvent market. In the example, UVEK just guessed that two production technologies were used equally for titantium dioxide, but the ecoinvent market has actual production volume numbers. In this case, we can use the market, but to best reflect the intent of the source database we need to remove the market transport. We could also shift over transport from a source process when appropriate.

Example: https://dare.cauldron.ch/compare/33242/17621

Actions: Delete rows or change amount. *Create proxy* button, and set match type to "Match with market and replace or remove transport". 

# 6. Match with market and replace or remove transport and loss factor

Same as number 5, but also use the source process loss factor.

Actions: Delete rows or change amount. *Create proxy* button, and set match type to "Match with market and replace or remove transport and loss factor". 

# 7. Equivalent datasets after modification

More modification than in 5/6 is needed to get equivalent processes, including deleting some target exchanges, shifting over some source exchanges, or expansion of target exchanges and further modification. Used only when actually necessary, this is dangerous surgery.

Example: https://dare.cauldron.ch/compare/31656/13237

Actions: Delete, shift, and change numbers in rows. *Create proxy* button, and set appropriate match type.

# 8. Replace aggregated with unit process

An older dataset which was aggregated for confidentiality (i.e. only had elementary flows) was replaced with 

Example: https://dare.cauldron.ch/compare/32551/34641

Actions: *Create proxy* button, and set match type to "Replace aggregated with unit process".

# 9. Rescaled direct correspondence

The source and target datasets are basically the same, but expressed in different units.

Example: https://dare.cauldron.ch/compare/31472/16158

Actions: *Collapse inputs* arrow, then click on the number 1.0 in the target process table to get the coefficient correct. *Create proxy* button, and set match type to "Rescaled direct correspondence".

How to get the correct value? The number you want is how much of the *target process* you need to provide one unit of the *source process*. So, if the source process was one litre of deionised water, but the target was one cubic meter, the coefficient would be 0.001 cubic meters per litre.

**Comment by marc-vdm** [comment]
What about https://dare.cauldron.ch/compare/31142/21942? 

I've added transport (type 4), but I also removed the waste disposal for the train here. To me that sounds like type 7, but it's not exactly 'extensive modification' either.

**Reply by cmutel** [reply]
Let's keep this as type 7, and I will change the label to reflect that this isn't always huge changes but more than present in the previous types.

**Comment by marc-vdm** [comment]
And for this: https://dare.cauldron.ch/compare/32641/11410

Does it make sense to add a new type (locations don't match, but data does) for this or should I keep that under type 2 (or even 1?)

**Reply by cmutel** [reply]
New match type added in 751e51ff81404550a86d251471402fa92b5f4bee.
